Platon/backend/app/file_system/views.py

```diff
@@ -0,0 +1,267 @@
+
+@file_system_ns.route("/folder")
+class FolderSystemAPI(Resource):
+
+    @api.doc(responses={
+                400: "Invalid Input",
+                401: "Authantication Problem",
+                404: "User if not found"
+    })
+    @api.response(200, 'Valid Folder', folder_models)
+    @api.expect(folder_get_parser)
+    @login_required
+    @workspace_exists(param_loc='args',workspace_id_key='workspace_id')
+    @active_contribution_required(param_loc='args',workspace_id_key='workspace_id')
+    def get(user_id,self):
+        form = FolderInfoForm(request.args)
+        if form.validate():
+            ws_path = FileSystem.workspace_base_path(form.workspace_id.data)
+            folder_path = ws_path + os.path.sep + form.path.data
+            # Control given path exits or not
+            if not FileSystem.is_directory_exists(folder_path):
+                return make_response(jsonify({"err":"Give Appropriate Path"}),400)
+            file_list = []
+            folder_list = []
+            # Find list of files and foldes in the given path
+            for element in os.listdir(folder_path):
+                if FileSystem.is_file_exists(folder_path + os.path.sep + element):
+                    file_list.append(element)
+                elif FileSystem.is_directory_exists(folder_path + os.path.sep + element):
+                    folder_list.append(element)
+            return make_response(jsonify({"files":file_list,"folders":folder_list,"cwd":form.path.data}),200)            
+        else:
+            return make_response(jsonify({"err":"Invalid Input"}),400)
+
+    @api.doc(responses={
+                201: "Folder Successfully Created",
+                400: "Invalid Input",
+                401: "Authantication Problem",
+                404: "User if not found"
+    })
+    @api.expect(folder_post_put_parser)
+    @login_required
+    @workspace_exists(param_loc='form',workspace_id_key='workspace_id')
+    @active_contribution_required(param_loc='form',workspace_id_key='workspace_id')
+    def post(user_id,self):
+        form = FolderPostPutForm(request.form)
+        if form.validate():
+            ws_path = FileSystem.workspace_base_path(form.workspace_id.data)
+            # Control given path exists or not
+            path = ws_path + os.path.sep + form.path.data
+            if not FileSystem.is_directory_exists(path):
+                return make_response(jsonify({"err":"Give Appropriate Path"}),400)
+            # Control new folder name is valid or not
+            new_folder_path = path + os.path.sep + form.new_folder_name.data
+            if FileSystem.is_directory_exists(new_folder_path):
+                return make_response(jsonify({"err":"Given Folder Exists"}),400)
+            # Create folder
+            try:
+                os.mkdir(new_folder_path)
+            except:
+                return make_response(jsonify({"err":"Recursive directories can not be created"}),400)
```

Comment:
how did you arrive to this conclusion? I didn't understand. Is it the only reason that causes line 201 to give error?

Reply:
By using os.mkdir(), it is impossible to create recursive directories. To do that, you have to open these folders one by one. os.makedirs() is the function that can create recursive directories on its own.